### ISSUE TITLE
Require incident disclosure transitively for all sub-CAs

### ISSUE THREAD

**Issue by sleevi** [mozilla/pkipolicy#193]
https://github.com/mozilla/pkipolicy/issues/168 (and the related fix, https://github.com/mozilla/pkipolicy/commit/5f1b0961fa66f824adca67d7021cd9c9c62a88fb ) track adding an overall requirement for CAs to provide timely incident reports.

One potential issue with that language is that it's ambiguous as to whether that transitively applies to third-party operated sub-CAs, and what the expectations are for root CAs. While it may be read that all CAs that directly or transitively chain to Mozilla-included roots are subject to the policy, the question is "Who is responsible" and "To what extent"

For example, consider Root CA "Root", which has signed a third-party operated intermediate, "Sub". If "Sub" discovers an issue, what is the expected reporting process?

Possible options:
1. Sub is responsible
  1. Sub reports to Root
  1. Sub reports to Auditor
  1. Sub opens an issue / creates a posting to Mozilla disclosing the incident
1. Root is responsible
  1. Sub reports to Root
  1. Sub reports to Auditor
  1. Root is responsible for opening an issue on behalf of Sub

If we go with (1), then there's a question of what happens if Sub provides sub-standard responses, or lacks timely responses, or even that Root knows what particular issue is reported and can be involved.

If we go with (2), then there's the issue that we're adding a middleman in the game of telephone, and responses can be both delayed and lack detail.

I think the expectation is:
1. Sub creates the issue
1. Sub is responsible for notifying Root of the issue
1. Root is responsible for following and engaging with the issue
1. Both Sub and Root are accountable for the responses, and their quality and detail

This is how it's worked out in practice for some incidents, but not all. 1.1 of the Policy 2.7 makes it clear the scope, but doesn't define the process/accountability for incident reports (5.3 makes the Root overall accountable).

To close the overall confusion, something in Section 2.1 (CA Operations) might be useful to make sure that CAs that sign certificates operated by other entities are responsible for ensuring these organizations comply with all requirements of the policy (e.g. explicitly stating it). Section 2.4 can make it clear that it's the Root's responsibility to ensure an Incident is filed and timely responded to, but they may allow the sub-CA to create the actual issue and engage on it.

**Comment by RufusJWB** [comment]
In the example of the relation between Quovadis (Root) and Siemens (Sub) we have established the following process after the Sub (we) discover / are made aware of an issue:

1) Sub reports to Root
2) Sub reports to Auditor (of Sub)
3) Root opens a bug / creates a posting to Mozilla disclosing the incident
4) Sub works on the issue under supervision of Root
5a) Root updates the bug based on information provided by Sub
5b) Sub updates the bug with status information

After fixing the bug:
1) Sub informs the Root
2a) Root closes the bug
2b) Sub closes the bug
3) Sub informs the Auditor (of Sub)

As a reference for this process see: https://bugzilla.mozilla.org/show_bug.cgi?id=1534535

For us this process does work very well

**Comment by sleevi** [comment]
Thanks @RufusJWB! I think that's a totally reasonable model and probably we want to make sure is explicit, and not simply implicit best practice.

I think the "Root opens the bug" could also be "Sub opens a bug / creates a posting"; it doesn't matter who, but the Root is responsible for making sure it happens and that 5a/5b happen.

**Comment by wthayer** [comment]
IMO, this is (policy-wise) simple - we hold the program member (i.e. root CA) accountable. We don't care who does what, as long as it happens. I'm happy to propose adding a sentence like "Root CAs are responsible for ensuring that these requirements are met for all certificates that directly or transitively chain to their certificates." to the fix for issue #168 to clarify the expectation.

**Comment by sleevi** [comment]
Yeah. I should have lead with the suggested change up front, then the motivating/practical examples.

I suppose my proposed changes to 2.1 are nominally covered by 1.1's scope. It's up to you whether it bears repeating in 2.1 that:

`
The CA whose certificates are included in Mozilla's root program MUST ...
...
7. Ensure that all certificates within the scope of this policy, as described in Section 1.1, adhere to this policy.
`

This makes it explicit that the scope of 1.1 flows into the Root CA's obligations.

I've struggled a little on how to further emphasize (explicitly) in 2.4 that the Root CA is responsible for ensuring reporting for sub-CAs; everything I tried ends up clunky, but I think it may be useful to make sure it's explicitly stated.
